src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -74,5 +76,121 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapHighWatermark(uint64_t & cu
 #endif
 }
 
+CHIP_ERROR DiagnosticDataProviderImpl::GetRebootCount(uint16_t & rebootCount)
+{
+    uint32_t count = 0;
+    CHIP_ERROR err = ConfigurationMgr().GetRebootCount(count);
+
+    if (err == CHIP_NO_ERROR)
+    {
+        VerifyOrReturnError(count <= UINT16_MAX, CHIP_ERROR_INVALID_INTEGER_VALUE);
+        rebootCount = static_cast<uint16_t>(count);
+    }
+
+    return err;
+}
+
+CHIP_ERROR DiagnosticDataProviderImpl::GetBootReason(uint8_t & bootReason)
+{
+    uint32_t reason = 0;
+
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    if (hwinfo_get_reset_cause(&reason) != 0)
+    {
+        return CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    if (reason & (RESET_POR | RESET_PIN))
+    {
+        bootReason = EMBER_ZCL_BOOT_REASON_TYPE_POWER_ON_REBOOT;
+    }
+    else if (reason & RESET_WATCHDOG)
+    {
+        bootReason = EMBER_ZCL_BOOT_REASON_TYPE_SOFTWARE_WATCHDOG_RESET;
+    }
+    else if (reason & RESET_BROWNOUT)
+    {
+        bootReason = EMBER_ZCL_BOOT_REASON_TYPE_BROWN_OUT_RESET;
+    }
+    else if (reason & (RESET_SOFTWARE | RESET_LOW_POWER_WAKE | RESET_CPU_LOCKUP | RESET_DEBUG))
```

Comment:
Hmm, low power, cpu lock-up and debug are not really software resets, right? It's not handled by the OS? I would classify those as UNSPECIFIED. @LuDuda thoughts?

Reply:
I also wasn't sure about it. That's why I decided to do so:
https://github.com/nrfconnect/sdk-hal_nordic/blob/main/nrfx/helpers/nrfx_reset_reason.h#L71
https://github.com/nrfconnect/sdk-hal_nordic/blob/main/nrfx/helpers/nrfx_reset_reason.h#L67

If it comes to the low power, I guess it's right I just read the wrong comment (they are below the name, not above):
https://github.com/nrfconnect/sdk-hal_nordic/blob/main/nrfx/helpers/nrfx_reset_reason.h#L73
According to that one it looks like hardware interrupt, so I think it is unspecified.